From 7b5743f65730334e5e9afa0ba8cff2368b417526 Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Tue, 11 Mar 2025 10:23:10 -0400
Subject: [PATCH] Issue 6665 - UI - Need to refresh log settings after saving

Description:

While do we reload the config from the parent component it is not resetting the
child component as expected.

Relates: https://github.com/389ds/389-ds-base/issues/6665

Reviewed by: spichugi(Thanks!)
---
 .../389-console/src/lib/server/accessLog.jsx  | 20 ++++++++----------
 .../389-console/src/lib/server/auditLog.jsx   | 21 ++++++++-----------
 .../src/lib/server/auditfailLog.jsx           | 20 ++++++++----------
 .../389-console/src/lib/server/errorLog.jsx   | 20 ++++++++----------
 .../src/lib/server/securityLog.jsx            | 20 ++++++++----------
 5 files changed, 45 insertions(+), 56 deletions(-)

diff --git a/src/cockpit/389-console/src/lib/server/accessLog.jsx b/src/cockpit/389-console/src/lib/server/accessLog.jsx
index c1e4bc4c4..803fa8795 100644
--- a/src/cockpit/389-console/src/lib/server/accessLog.jsx
+++ b/src/cockpit/389-console/src/lib/server/accessLog.jsx
@@ -301,32 +301,30 @@ export class ServerAccessLog extends React.Component {
                 .spawn(cmd, { superuser: true, err: "message" })
                 .done(content => {
                     this.props.reload();
+                    this.refreshConfig(1);
                     this.props.addNotification(
                         "success",
                         _("Successfully updated Access Log settings")
                     );
-                    this.setState({
-                        loading: false
-                    });
                 })
                 .fail(err => {
                     const errMsg = JSON.parse(err);
                     this.props.reload();
+                    this.refreshConfig(1);
                     this.props.addNotification(
                         "error",
                         cockpit.format(_("Error saving Access Log settings - $0"), errMsg.desc)
                     );
-                    this.setState({
-                        loading: false
-                    });
                 });
     }
 
-    refreshConfig(refesh) {
-        this.setState({
-            loading: true,
-            loaded: false,
-        });
+    refreshConfig(loading) {
+        if (!loading) {
+            this.setState({
+                loading: true,
+                loaded: false,
+            });
+        }
 
         const cmd = [
             "dsconf", "-j", "ldapi://%2fvar%2frun%2fslapd-" + this.props.serverId + ".socket",
diff --git a/src/cockpit/389-console/src/lib/server/auditLog.jsx b/src/cockpit/389-console/src/lib/server/auditLog.jsx
index c4a55008c..40f681e93 100644
--- a/src/cockpit/389-console/src/lib/server/auditLog.jsx
+++ b/src/cockpit/389-console/src/lib/server/auditLog.jsx
@@ -308,33 +308,30 @@ export class ServerAuditLog extends React.Component {
                 .spawn(cmd, { superuser: true, err: "message" })
                 .done(content => {
                     this.props.reload();
+                    this.refreshConfig(1);
                     this.props.addNotification(
                         "success",
                         _("Successfully updated Audit Log settings")
                     );
-                    this.setState({
-                        loading: false
-                    });
-
                 })
                 .fail(err => {
                     const errMsg = JSON.parse(err);
                     this.props.reload();
+                    this.refreshConfig(1);
                     this.props.addNotification(
                         "error",
                         cockpit.format(_("Error saving Audit Log settings - $0"), errMsg.desc)
                     );
-                    this.setState({
-                        loading: false,
-                    });
                 });
     }
 
-    refreshConfig() {
-        this.setState({
-            loading: true,
-            loaded: false,
-        });
+    refreshConfig(loading) {
+        if (!loading) {
+            this.setState({
+                loading: true,
+                loaded: false,
+            });
+        }
 
         const cmd = [
             "dsconf", "-j", "ldapi://%2fvar%2frun%2fslapd-" + this.props.serverId + ".socket",
diff --git a/src/cockpit/389-console/src/lib/server/auditfailLog.jsx b/src/cockpit/389-console/src/lib/server/auditfailLog.jsx
index 0e5ae0a88..19785276a 100644
--- a/src/cockpit/389-console/src/lib/server/auditfailLog.jsx
+++ b/src/cockpit/389-console/src/lib/server/auditfailLog.jsx
@@ -244,32 +244,30 @@ export class ServerAuditFailLog extends React.Component {
                 .spawn(cmd, { superuser: true, err: "message" })
                 .done(content => {
                     this.props.reload();
+                    this.refreshConfig(1);
                     this.props.addNotification(
                         "success",
                         _("Successfully updated Audit Fail Log settings")
                     );
-                    this.setState({
-                        loading: false
-                    });
                 })
                 .fail(err => {
                     const errMsg = JSON.parse(err);
                     this.props.reload();
+                    this.refreshConfig(1);
                     this.props.addNotification(
                         "error",
                         cockpit.format(_("Error saving Audit Fail Log settings - $0"), errMsg.desc)
                     );
-                    this.setState({
-                        loading: false
-                    });
                 });
     }
 
-    refreshConfig() {
-        this.setState({
-            loading: true,
-            loaded: false,
-        });
+    refreshConfig(loading) {
+        if (!loading) {
+            this.setState({
+                loading: true,
+                loaded: false,
+            });
+        }
 
         const cmd = [
             "dsconf", "-j", "ldapi://%2fvar%2frun%2fslapd-" + this.props.serverId + ".socket",
diff --git a/src/cockpit/389-console/src/lib/server/errorLog.jsx b/src/cockpit/389-console/src/lib/server/errorLog.jsx
index 0922a80ff..0ad36e594 100644
--- a/src/cockpit/389-console/src/lib/server/errorLog.jsx
+++ b/src/cockpit/389-console/src/lib/server/errorLog.jsx
@@ -326,32 +326,30 @@ export class ServerErrorLog extends React.Component {
                 .spawn(cmd, { superuser: true, err: "message" })
                 .done(content => {
                     this.props.reload();
+                    this.handleRefreshConfig(1);
                     this.props.addNotification(
                         "success",
                         _("Successfully updated Error Log settings")
                     );
-                    this.setState({
-                        loading: false
-                    });
                 })
                 .fail(err => {
                     const errMsg = JSON.parse(err);
                     this.props.reload();
+                    this.handleRefreshConfig(1);
                     this.props.addNotification(
                         "error",
                         cockpit.format(_("Error saving Error Log settings - $0"), errMsg.desc)
                     );
-                    this.setState({
-                        loading: false
-                    });
                 });
     }
 
-    handleRefreshConfig() {
-        this.setState({
-            loading: true,
-            loaded: false,
-        });
+    handleRefreshConfig(loading) {
+        if (!loading) {
+            this.setState({
+                loading: true,
+                loaded: false,
+            });
+        };
 
         const cmd = [
             "dsconf", "-j", "ldapi://%2fvar%2frun%2fslapd-" + this.props.serverId + ".socket",
diff --git a/src/cockpit/389-console/src/lib/server/securityLog.jsx b/src/cockpit/389-console/src/lib/server/securityLog.jsx
index 01adcca0c..77000a873 100644
--- a/src/cockpit/389-console/src/lib/server/securityLog.jsx
+++ b/src/cockpit/389-console/src/lib/server/securityLog.jsx
@@ -245,32 +245,30 @@ export class ServerSecurityLog extends React.Component {
                 .spawn(cmd, { superuser: true, err: "message" })
                 .done(content => {
                     this.props.reload();
+                    this.refreshConfig(1);
                     this.props.addNotification(
                         "success",
                         _("Successfully updated Security Log settings")
                     );
-                    this.setState({
-                        loading: false
-                    });
                 })
                 .fail(err => {
                     const errMsg = JSON.parse(err);
                     this.props.reload();
+                    this.refreshConfig(1);
                     this.props.addNotification(
                         "error",
                         cockpit.format(_("Error saving Security Log settings - $0"), errMsg.desc)
                     );
-                    this.setState({
-                        loading: false
-                    });
                 });
     }
 
-    refreshConfig(refesh) {
-        this.setState({
-            loading: true,
-            loaded: false,
-        });
+    refreshConfig(loading) {
+        if (!loading) {
+            this.setState({
+                loading: true,
+                loaded: false,
+            });
+        }
 
         const cmd = [
             "dsconf", "-j", "ldapi://%2fvar%2frun%2fslapd-" + this.props.serverId + ".socket",
-- 
2.48.1

